### PR TITLE
osm-cli: init at 0.1.0

### PR DESCRIPTION
**Pull request by lucperkins**
###### Motivation for this change

Added a package for the [Open Service Mesh](https://openservicemesh.io/) CLI tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/osm/cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+let
+  version = "0.1.0";
+  sha256 = "0hc63xvww4p4bqsq2ihz5in5ia9x7ibfa7ixd6fkxvdb1rbwxrx6";
+  vendorSha256 = "03qy4gl9lii4amsfg2y4ib5xwg74gri601pkfa3c7jwc034xf5kf";
+in buildGoModule {
+  name = "osm-cli-${version}";
+  pname = "osm";
+  version = version;
+
+  vendorSha256 = vendorSha256;
```
Comment (danieldk) [inline pkgs/applications/networking/cluster/osm/cli/default.nix:12]:
```suggestion
  inherit version vendorSha256;

  pname = "osm";
```

`name` is defined in terms of `pname` and `version` in `stdenv`, so you do not need to define it explicitly.

File: pkgs/applications/networking/cluster/osm/cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "openservicemesh";
+    repo = "osm";
+    rev = "v${version}";
+    sha256 = sha256;
```
Comment (danieldk) [inline pkgs/applications/networking/cluster/osm/cli/default.nix:18]:
```suggestion
    inherit sha256;
    owner = "openservicemesh";
    repo = "osm";
    rev = "v${version}";
```

**Review by danieldk — CHANGES_REQUESTED** [review]

**Comment by lucperkins** [comment]
@danieldk Excellent suggestions! I have made those updates. Thank you for your speedy and thorough review :100:

**Comment by danieldk** [comment]
One more question, I saw that the derivation is added to `pkgs/applications/networking/cluster/osm/cli` rather than `pkgs/applications/networking/cluster/osm` or `pkgs/applications/networking/cluster/osm-cli`. Any particular reason?

**Comment by lucperkins** [comment]
@danieldk I arranged it that way because I think there may ultimately be multiple subpackages under this project, such as the [controller](https://github.com/openservicemesh/osm/tree/main/cmd/osm-controller).

**Comment by danieldk** [comment]
> @danieldk I arranged it that way because I think there may ultimately be multiple subpackages under this project, such as the [controller](https://github.com/openservicemesh/osm/tree/main/cmd/osm-controller).

I see. Wouldn't it make more sense to have a single package `osm` that installs both the controller and CLI, since they are part of the same source package?

File: pkgs/applications/networking/cluster/osm/cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+in buildGoModule {
+  inherit pname version vendorSha256;
+
+  src = fetchFromGitHub {
+    inherit sha256;
+
+    owner = "openservicemesh";
+    repo = "osm";
+    rev = "v${version}";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/osm/cli/default.nix:17]:
```suggestion
buildGoModule rec {
  pname = "osm-cli";
  version = "0.1.0";

  src = fetchFromGitHub {
    owner = "openservicemesh";
    repo = "osm";
    rev = "v${version}";
    sha256 = "0hc63xvww4p4bqsq2ihz5in5ia9x7ibfa7ixd6fkxvdb1rbwxrx6";
  };

  vendorSha256 = "03qy4gl9lii4amsfg2y4ib5xwg74gri601pkfa3c7jwc034xf5kf";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 94748` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>osm-cli</li>
  </ul>
</details>

**Pull request closed without merging**